<!DOCTYPE html>
<html lang="en">

<head>

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-1QG2TZS458"></script>
  <script>
    window.dataLayer = window.dataLayer || [];

    function gtag() {
      dataLayer.push(arguments);
    }
    gtag('js', new Date());

    gtag('config', 'G-1QG2TZS458');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta name="description" content="Website for Jordan A. Berg, Biochemistry Ph.D. student at the University of Utah">
  <meta name="keywords" content="Jordan, Berg, Biochemistry, Utah, Computational, Biology, Metabolism">
  <meta name="author" content="Jordan Berg">
  <meta name="google-site-verification" content="7R5dCQpyVk7BtHBQhqaXIUbCffeulxE92qqUJDMj7nM" />

  <!-- Page Title -->
  <title>Network Metabolism Group - Blog</title>
  <!-- / -->

  <!---Font Icon-->
  <link href="static/plugin/font-awesome/css/fontawesome-all.min.css" rel="stylesheet">
  <link href="static/plugin/themify-icons/themify-icons.css" rel="stylesheet">
  <!-- / -->

  <!-- Plugin CSS -->
  <link href="static/plugin/bootstrap/css/bootstrap.min.css" rel="stylesheet">
  <link href="static/plugin/owl-carousel/css/owl.carousel.min.css" rel="stylesheet">
  <link href="static/plugin/magnific/magnific-popup.css" rel="stylesheet">
  <!-- / -->

  <!-- Theme Style -->
  <link href="static/css/styles.css" rel="stylesheet">
  <link href="static/css/color/color-7.css" rel="stylesheet" id="color_theme">
  <!-- / -->

  <!-- Favicon -->
  <link rel="icon" href="static\img\Noun_project_network_icon_1365244_cc.svg" />
  <!-- / -->

  <!-- Custom Style -->
  <link href="static/css/custom.css" rel="stylesheet">
  <!-- / -->

</head>

<!-- Body Start -->

<body data-spy="scroll" data-target="#navbarRyan" data-offset="98">

  <script src="https://d3js.org/d3.v5.min.js"></script>

  <!-- Loading -->
  <div id="loading">
    <div class="load-circle"><span class="one"></span></div>
  </div>
  <!-- / -->

  <!-- Header -->
  <header>
    <nav class="navbar header-nav header-nav-white fixed-top navbar-expand-lg">
      <div class="container-fluid">
        <!-- Brand -->
        <a class="navbar-brand font-alt" href="index.html">Network Metabolism Group</a>
        <!-- / -->

        <!-- Mobile Toggle -->
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarRyan" aria-controls="navbarRyan" aria-expanded="false" aria-label="Toggle navigation">
          <span></span>
          <span></span>
          <span></span>
        </button>
        <!-- / -->

        <!-- Top Menu -->
        <div class="collapse navbar-collapse justify-content-end" id="navbarRyan">
          <ul class="navbar-nav">
            <li><a class="nav-link" href="index.html">About</a></li>
            <li><a class="nav-link" href="research.html">Research</a></li>
            <li><a class="nav-link" href="publications.html">Publications</a></li>
            <li><a class="nav-link" href="people.html">People</a></li>
            <li><a class="nav-link" href="funding.html">Funding</a></li>
            <li><a class="nav-link" href="blog.html">Blog</a></li>
            <li><a class="nav-link" href="contact.html">Contact</a></li>
            <li></li>
            <li></li>
          </ul>
        </div>
        <!-- / -->

      </div><!-- Container -->
    </nav> <!-- Navbar -->
  </header>
  <!-- Header End -->


  <!-- Blog Start -->
  <main class="gray-bg">
    <section id="blog" class="section">
      <div class="container">
        <div class="row justify-content-center m-60px-b md-m-30px-b">
          <div class="col-12 col-md-10 col-lg-7">
            <div class="section-title text-center">
              <h2 class="font-alt">Blog</h2>
            </div>
          </div>
        </div>
        <div class="row md-m-30px-b md-m-10px-r md-m-10px-l">

          <!-- BLOG 3 START -->
          <div id="blog-3" class="col-12 m-60px-b md-m-30px-b">

            <a href="" class="p-btn" data-toggle="modal" data-target=".blog-3-modal">
              <h4>Getting an F99/K00 Fellowship</h4>
            </a>
            <p>(14 Jun 2021) My tips and tricks for the F99/K00 fellowship (whatever they're worth).</p>

            <!-- blog-3-modal start  -->
            <div class="modal fade blog-3-modal" tabindex="-1" role="dialog" aria-labelledby="myLargeModalLabel" aria-hidden="true">
              <div class="modal-dialog modal-lg modal-dialog-scrollable">
                <div class="modal-content">
                  <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
                      <span aria-hidden="true">&times;</span>
                    </button>

                    <span class="post-meta"><b>14 Jun 2021</b>
                      <h4 class="modal-title" id="exampleModalScrollableTitle">Getting an F99/K00 Fellowship</h4>

                  </div>
                  <div class="modal-body">

                    <b>Disclaimer:</b> Obviously, this is all based on my experience. Results may vary.
                    <br><br>
                    Access my completed application <a href="blog/resources/2021-06-12_f99/Compiled_Application_Redacted.pdf" target="_blank">here</a>.
                    <br>
                    Access my institutional Letter of Intent <a href="blog/resources/2021-06-12_f99/Berg_LOI_F99_K00_Redacted.pdf" target="_blank">here</a>. 
                    <br><br><br>

                    <h5>
                      What is an F99/K00 Fellowship?
                    </h5>
                    <p>
                      From the RFA I used for my application:
                    </p>
                    <blockquote class="custom-blockquote">
                      The purpose of the NCI Predoctoral to Postdoctoral Fellow Transition Award (F99/K00) is to encourage and retain outstanding graduate students recognized 
                      by their institutions for their high potential and strong interest in pursuing careers as independent cancer researchers. The award will facilitate the 
                      transition of talented graduate students into successful cancer research postdoctoral appointments, and provide opportunities for career development 
                      activities relevant to their long-term career goals of becoming independent cancer researchers.
                    </blockquote>

                    <b>What RFA I used:</b> <a href="https://grants.nih.gov/grants/guide/rfa-files/RFA-CA-19-057.html" target="_blank">RFA-CA-19-057</a>
                    <br><br><br>

                    <h5>
                      My Previous Experience Applying for Fellowships
                    </h5>
                    <p>
                      Before I applied for the NCI’s F99/K00 fellowship, I had previously applied for the NSF <a href="https://www.nsfgrfp.org/" target="_blank">GRFP</a> and 
                      an NIH NIDDK F31, both during my 2nd year of grad school. Both applications were unsuccessful. I think a few of consistent pieces of feedback I got for both applications were:
                    </p>
                    <ol>
                      <li>
                        The research ideas were good but not as thoroughly thought out like they would have liked to see.
                      </li>
                      <li>
                        In each application, my undergraduate grades were brought up. This was a frustrating point to me because during my undergraduate studies, I was very involved in lab research 
                        and enjoyed the research aspects much more than actual classwork. As a result, my grades suffered (many C’s, failed OChem). I had hoped that the fact that I had two first-author 
                        papers from undergrad would overcome my poor grades, but no. One important point that my advisor made though in response to this, was that once I got a publication out, reviewers 
                        would stop bringing up my grades (which is true).
                      </li>
                      <li>
                        The reviewers were not impressed with my training and career development plans.
                      </li>
                      <li>
                        The reviewers were concerned about my advisor’s ability to provide mentoring due to their busy schedule.
                      </li>
                    </ol>
                    <p>
                      While these were not successful, I do think that applying for these before the F99 was important and helpful. Writing an F-series fellowship application is a huge undertaking and given 
                      the timeline I was under for the F99 application, it would have been crazy to be writing my first F-series fellowship application for the F99. And for context, my impact score was 53 on 
                      my original NIDDK F31 application. During the time between my unsuccessful GRFP and F31, I did successfully get funding through an NIDDK T32, with the focus being “Computational Approaches 
                      to Diabetes and Metabolism.
                    </p>
                    <br><br>

                    <h5>
                      Getting Started: Submitting my Institutional Letter of Intent
                    </h5>
                    <p>
                      I had previously discussed with my advisor my interest in applying for the F99 fellowship, especially as I had a friend in grad school who was awarded the F99 two years before, and a post-doc 
                      in our lab was on the K00 portion of their fellowship. For whatever reason, I did not find out about the institutional competition for the F99 application until Nov 7, 2019, and the internal 
                      application was due Nov 14. Whoever was selected then had to submit their full F99 application by Dec 4th (!!! – also, it was a good thing I had already applied for an F-series fellowship before 
                      and had started a new F31 application a few months before). Luckily, the internal application was pretty straight-forward: my 
                      <a href="blog/resources/2021-06-12_f99/Berg_Biosketch_Nov_2019_Redacted.pdf" target="_blank">NIH Biosketch</a> and my 
                      <a href="blog/resources/2021-06-12_f99/Berg_LOI_F99_K00_Redacted.pdf" target="_blank">Letter of Intent</a>.
                    </p>
                    <p>
                      By this point, I had already completed my first publication from grad school, which by this point had been posted on <i>bioRxiv</i> and we had just submitted our revisions to the journal. I had also 
                      participated in a couple of collaborative projects with a couple of resulting projects that were either already published or undergoing revisions with a journal. In my Letter of Intent, I made 
                      sure to emphasize these publications. Another big point I tried to sell was how my research was interdisciplinary – where I utilized mostly computational approaches to solving metabolic questions, 
                      but at the same time collaborated closely with a lot of wet lab biologists. At the time, I believe the NCI had recently announced some of their dedicated initiatives for Big Data in Cancer Research, 
                      which was another point I tried to emphasize here.
                    </p>
                    <br><br>

                    <h5>
                      Writing my Application
                    </h5>
                    <p>
                      On Nov 15th, I received the news that I had been selected to be the University of Utah/Huntsman Cancer Institute’s applicant for that cycle. Which was exciting, yet horrifying, since that meant I had 
                      just about 3 weeks to get everything together and submit my application. On top of all of this, I was going to be out of town Nov 18th - Dec 2nd for a conference and then a vacation right after with 
                      my partner and her parents. Fortunately, my original plan had been to apply again for an F31 during the Fall of 2019, so I had already begun this application to some extent.
                    </p>
                    <p>
                      Immediately after I found out I was selected, I emailed my letter writers for letters of recommendation and reached out to the post-doctoral fellow in our lab and the other friend, and they were gracious 
                      enough to send me their F99 applications to use as guidance. I had also previously read Jean Fan’s excellent <a href="https://jef.works/blog/2018/10/31/NIH-F99-K00-grant-tips-example-and-personal-experience/" target="_blank">blog post</a> 
                      outlining tips and tricks for the F99 application. Additionally, Dr. Fan’s application can be found on the linked blog post, as well as a link to <a href="https://www.ogrants.org/" target="_blank">Open Grants</a>.
                    </p>
                    <p>
                      Another initial step I took when I started writing my F99 application was to re-read the reviews from my previous unsuccessful F31 application. As I already outlined above, there were some particular points 
                      of feedback that stood out to me from my previous application and, luckily, point #2 was taken care of as I had a first author publication in the works. To work on the remaining points listed above, I focused 
                      on three “lines of attack” for preparing my application:
                    </p>
                    <ol>
                      <li>
                        <b>Emphasize points of interest for the funding agency</b> 
                        <p>
                          Obviously, it needs to be clear to the funding agency that funding your research will advance their goals. To address this point, I tried to emphasize in each document (cover letter, research plan, 
                          training plan, etc.) that my research was interdisciplinary and would produce valuable tools and research that would help us better understand the complex systems of cancer metabolism. I also made 
                          sure to emphasize that while cancer metabolism is a very established field, a huge hurdle to accelerating advances was a lack of computational tools for analyzing these data, and pointed out a semi-recent 
                          review by Ralph DeBerardinis and Navdeep Chandel (Sci Adv, 2016) () that made this point clear. I also made sure to mention my co-mentor’s expertise in computer science and network modeling in nearly every 
                          section of my application (take-away: any points you want to emphasize in your application should be mentioned in every section of your application where appropriate).
                        </p>
                        <p>
                          Another point I wanted to emphasize based on previous feedback was my mentor’s ability to provide mentoring based on the “large group of trainees working in the lab”. This point I tried to emphasize mainly 
                          in my training plan. 
                        </p>
                        <blockquote class="custom-blockquote">
                          I am extremely fortunate to work in Dr. Rutter’s lab, where I am given a great amount of trust and independence with my project and the its direction. I meet with Dr. Rutter one-on-one once a week to discuss 
                          ideas and experimental plans, and also frequently in person or via email other times throughout the week. Dr. Rutter has always provided prompt feedback that has been invaluable as I develop my projects. Dr. 
                          Rutter has truly become a mentor in every sense of the term, and someone I have learned from immeasurably. I also meet with the Rutter lab Big Data group every week to discuss the ideas proposed in this 
                          application and their progress. I am also able to provide feedback for my peers in this venue. I present my research often in lab meeting to get feedback on the direction of my projects. These meetings have 
                          been consequential to the development of my ideas as the members of the lab come from diverse backgrounds and have valuable perspectives. Additionally, I have cultivated a network of computational biologists 
                          at the University of Utah and often reach out to them with questions. The University of Utah has a strong collaborative spirit, therefore when my established network is unable to provide the necessary guidance, 
                          it is easy to reach out to someone with the necessary expertise at the University and they have always welcomed these interactions. For example, before presenting my research to a more computational audience, 
                          I meet with this network to hone my presentation for scientists from a computational background. Not only do these interactions help improve the research that I perform, but they also provide me with training 
                          to present my ideas and results in a concise and clear manner and defend those ideas in front of experienced graduate students, postdoctoral fellows, and senior scientists.
                        </blockquote>
                        <p>
                          I also tried to emphasize my collaborator network and the feedback I have been able to receive from them to supplement the training from my advisor.
                        </p>
                      </li>
                      <li>
                        <b>Organization/readability</b>
                        <p>
                          Another theme from my previous application that I felt was common was the lack of clarity. Sure, I thought I had been clear, but I know what I am thinking and the context behind my thoughts, so obviously what 
                          I thought was clear was not. In looking at a friend’s successful F99 application, one striking pattern I saw in their application was the usage of tables and sub-section headers. With this in mind, I tried to 
                          make my Training Plan very organized and succinct, using headers such as “Research Development”, “Computational Training”, “Science Communication”, “Leadership”. I feel like this helped my overall organization 
                          of thoughts and ideas, and even if something was not totally clear to a reviewer, the theme was there underlined and in bold. I also included a table of my timeline and the milestones I wanted to reach during 
                          my training. I maybe also overused bolded text, but I do think it really helps emphasize the important points I wanted to make in my application.
                        </p>
                      </li>
                      <li>
                        <b>Feedback</b>
                        <p>
                          The final theme I want to emphasize in preparing my application was getting feedback. This was particularly important given the accelerated timeline on I was on in preparing the application. In all, I think 
                          I remember three people giving me feedback on my Research Plan: 1) the post-doc in our lab on the K00-phase of their fellowship, 2) a fellow graduate student who was most familiar with my research and could 
                          make sure the key tenets of my project were clear and well-thought-out, and 3) my partner, who coincidentally I met in the Rutter lab.
                        </p>
                      </li>
                      <li>
                        <b>Other</b>
                        <p>
                          Another interesting point of concern in preparing my application was that I might be emphasizing my collaborative project too much. I think luckily more and more researchers are recognizing the importance 
                          of collaborative research more and placing less emphasis on “genius science” (I say this sarcastically, but what I mean are CNS papers with few authors). My advisor let me know though that this is potentially 
                          a point some reviewer could view as a weakness. Thus, I made sure to emphasize this point in my Training Plan with the following:
                        </p>
                        <blockquote class="custom-blockquote">
                          In addition to the papers I will publish as first author, and to make full use of my unique background in computational metabolism, I plan to work on 2-4 collaboration projects per year that will lead to 
                          additional coauthor papers. While I have been able to contribute significantly to each of these projects, it has not deterred me from maintaining a high level of productivity on my personal projects.
                        </blockquote>
                        <p>
                          And,
                        </p>
                        <p>
                          Make sure you follow the Fellowship (F) instructions in the <a href="https://grants.nih.gov/grants/how-to-apply-application-guide.html" target="_blank">SF424 (R&R) Application Guide</a>. Read through every 
                          section and make sure you have formatted everything correctly, included all the information asked for, and so on!
                        </p>
                        <p>
                          Hopefully, these points are useful and give an alternative perspective to someone considering applying for an F99 fellowship. Particularly, it is low-cost to write a Letter of Intent for your institutional 
                          competition (assuming that is all they require in addition to your biosketch) and if you are selected, there is a relatively reasonable chance to be awarded the fellowship since you are only competing against 
                          one possible applicant from (in the case of the NCI) around 70 <a href="https://www.cancer.gov/research/infrastructure/cancer-centers" target="_blank">NCI-Designated Cancer Centers</a>. 
                        </p>
                      </li>
                    </ol>
                    <p>
                      You can access my application <a href="blog/resources/2021-06-12_f99/Compiled_Application_Redacted.pdf" target="_blank">here</a>.
                    </p>
                    <br><br>

                    <h5>
                      Application Feedback and Waiting for the Final Notice
                    </h5>
                    <p>
                      So, long story short, I submitted my application in December 2019 and then waited. And waited. And waited. Also, a few months after my application submission, the COVID pandemic hit, and so then I had questions 
                      like “are they going to cut funding?”, “is this going to complicate the review?”, etc. I probably checked eRA commons hundreds of times. Finally, reviews were released April 22, 2020 and I received my Notice of 
                      Award July 22, 2020 with a start date of Aug 1, 2020. And to be honest, I don’t know if this is the general timeline (like getting no other indication of the award until the NOA a week before the start date) or 
                      if this was a delayed timeline due to all things COVID.
                    </p>
                    <p>
                      For my F99 application, I got an impact score of 19 (a big improvement from 53 from my previous F31!). It definitely wasn't a perfect application, and some of the same issues I ran into earlier were mentioned, but 
                      but instead usually only one reviewer would mention a given weakness and didn't seem to emphasize the point too much.
                    </p>
                    <br><br>

                    <h5>
                      Overall Timeline
                    </h5>
                    <b>Institute LOI Deadline</b>: 11/14/2019
                    <br>
                    <b>Application Submission</b>: 12/04/2019
                    <br>
                    <b>Reviews Recieved</b>: 04/22/2020
                    <br>
                    <b>NOA</b>: 7/22/2020
                    <br>
                    <b>Start Date</b>: 08/01/2020
                    <br><br><br>

                    <h5>
                      Conclusion
                    </h5>
                    <p>
                      I know this is a case of survivor bias, but you should apply! It is definitely worth it, and you never know what will happen! I definitely thought it was a long-shot, but I'm glad I did it. And I know a lot of
                      people with F99's that had previously received an F31 beforehand, but that is not always the case! I definitely didn't have one, and I know of others personally who didn't have one beforehand either!
                    </p>
                    <p>
                      Let me know if you have any questions! I am happy to answer application-related questions, and, based on schedule, review parts of your application if you decide to apply and want some feedback. 
                      You can <a href = "mailto: jordan.berg@biochem.utah.edu">email me</a> (maybe add a subject line of “F99 question” or something to make sure it doesn’t get lost in my inbox), and feel 
                      free to pester me if you don’t hear back.
                    </p>


                    <br>
                    <br>
                  </div>
                  <div class="modal-footer d-flex justify-content-between">
                    <button type="button" class="btn" data-dismiss="modal">Close</button>
                    <div align="right">
                      <a href="https://github.com/j-berg/j-berg.github.io/tree/main/blog/code/2020-02-12_superspreading" target="_blank" class="m-btn">Code</a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <!-- project-3-modal end -->
          </div>
          <!-- PROJECT 3 END -->
          <br>


          <!-- BLOG 2 START -->
          <div id="blog-2" class="col-12 m-60px-b md-m-30px-b">

            <a href="" class="p-btn" data-toggle="modal" data-target=".blog-2-modal">
              <h4>Interactive Article Exploration: "Previously undetected super-spreading of <i>Mycobacterium tuberculosis</i> revealed by deep sequencing</h4>
            </a>
            <p>(14 Feb 2020) A re-visualization of infection spread in a tuberculosis community outbreak.</p>

            <!-- blog-2-modal start  -->
            <div class="modal fade blog-2-modal" tabindex="-1" role="dialog" aria-labelledby="myLargeModalLabel" aria-hidden="true">
              <div class="modal-dialog modal-lg modal-dialog-scrollable">
                <div class="modal-content">
                  <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
                      <span aria-hidden="true">&times;</span>
                    </button>

                    <span class="post-meta"><b>14 Feb 2020</b>
                      <h4 class="modal-title" id="exampleModalScrollableTitle">Interactive Article Exploration: "Previously undetected super-spreading of <i>Mycobacterium tuberculosis</i> revealed by deep sequencing</h4>

                  </div>
                  <div class="modal-body">

                    <p>* Find the original paper <a href="https://elifesciences.org/articles/53245" target="_blank">here</a> [1].</p>
                    <p>* Find the code <a href="https://github.com/j-berg/j-berg.github.io/tree/main/blog/code/2020-02-12_superspreading" target="_blank">here</a>.</p>

                    <p>Earlier this month, a paper was published in eLife detailing the deep sequencing of several tuberculosis patients from an outbreak within a Canadian Inuit community. By performing heirarchal clustering of the SNP data, the
                      authors were able to determine the presence of a "super-spreader" during the outbreak. Reading the paper, I became interested in trying to create an interactive visualization of the data that accounted for sequence similarity of
                      the tuberculosis strains isolated from the patients, as well as time derived from the month and year of diagnosis.
                    </p>

                    <p>In order to map spread, I took the CI 95% SNP data and created a square-form similarity matrix by using a Hamming distance calculation [<a href="https://www.nature.com/articles/srep10247" target="_blank">2</a>]. I then
                      generated a two-dimensional network, where nodes are connected by time and SNP similarity. To attempt to model the spread of the disease, I traversed time and found each patients most similar SNP neighbors, assuming that each
                      patient was infected when diagnosed, and that SNP variation indicates progression of the disease. Progression using this paradigm seemed to be most stable when for each iteration, I found the 3 minimum distance scores and linked
                      any patient with those distance values compared to the current patient of interest. Once a patient had been connected, they were removed from further analysis. Again, this probably over-simplifies the problem, but an interesting
                      approach nonetheless (hopefully).

                    <p>Disclaimer: It is interesting that while super-spreaders are identified, they are different than those identified in the original study. However, this was mostly a learning exercise for visualization of temporal data. As
                      compelling as this view of the data might be, it is preliminary and probably riddled with errors. Take this with a grain of salt...
                    </p>

                    <p>In order to interact with the graph, you can click a patient circle and drag to reposition the graph. To pan, click and drag the background. To zoom in and out, use the scroll wheel. To move through time, drag the ball on the
                      timeline below the graph, or press play or pause. Red coloring of the patient circle indicates time of diagnosis. As you can see, there is a trend/correlation with SNP similarity and time of diagnosis. Also, the reference genome
                      was excluded from the graph. Note: The two outlier strains, as identified in the original study, usually appear by themselves on the lower right corner of the graph, so you may need to pan over to view those samples.
                    </p>

                    <link rel="stylesheet" href="blog/code/2020-02-12_superspreading/tuberculosis_hold.css">
                    <div id="_border">
                      <div id="graph" align="center"></div>
                    </div>
                    <div id="play-button" align="center">Pause</div>
                    <div id="bar"><br>
                    <h4 align="center" style="font-family: sans-serif;">Time at Diagnosis (in months)</h4>
                    </div>
                    <script src="blog/code/2020-02-12_superspreading/tuberculosis_hold.js"></script>

                    <br><br>
                    <h3>References</h3>
                    1. Lee SL, et. al. Previously undetected super-spreading of <i>Mycobacterium tuberculosis</i> revealed by deep sequencing. (2020) <b><i>Elife.</i></b>
                    <br>
                    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                    <font size="2">
                      <a href="https://elifesciences.org/articles/53245" target="_blank"><i>Publication</i></a>
                      &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                    </font>
                    <br>
                    <br>
                    2. Hu H, et. al. Evaluating information content of SNPs for sample-tagging in re-sequencing projects. (2015) <b><i>Scientific Reports.</i></b> <br>
                    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                    <font size="2">
                      <a href="https://www.nature.com/articles/srep10247" target="_blank"><i>Publication</i></a>
                      &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                    </font>
                    <br>
                    <br>

                  </div>
                  <div class="modal-footer d-flex justify-content-between">
                    <button type="button" class="btn" data-dismiss="modal">Close</button>
                    <div align="right">
                      <a href="https://github.com/j-berg/j-berg.github.io/tree/main/blog/code/2020-02-12_superspreading" target="_blank" class="m-btn">Code</a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <!-- project-2-modal end -->
          </div>
          <!-- PROJECT 2 END -->
          <br>


          <!-- BLOG 1 START -->
          <div id="blog-1" class="col-12 m-60px-b md-m-30px-b">

            <a href="" class="p-btn" data-toggle="modal" data-target=".blog-1-modal">
              <h4>Introducing: XPRESSyourself -- an automated pipeline for ribosome profiling analysis</h4>
            </a>
            <p>(19 Jan 2020) A discussion of the software package XPRESSyourself, which introduces new tools for ribosome profiling and acts as a reference pipeline for ribosome profiling data analysis.</p>

            <!-- blog-0-modal start  -->
            <div class="modal fade blog-1-modal" tabindex="-1" role="dialog" aria-labelledby="myLargeModalLabel" aria-hidden="true">
              <div class="modal-dialog modal-lg modal-dialog-scrollable">
                <div class="modal-content">
                  <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
                      <span aria-hidden="true">&times;</span>
                    </button>

                    <span class="post-meta"><b>19 Jan 2020</b>
                      <h4 class="modal-title" id="exampleModalScrollableTitle">Introducing: XPRESSyourself -- an automated pipeline for ribosome profiling analysis</h4>

                  </div>
                  <div class="modal-body">
                    <p>* Find the software <a href="https://github.com/XPRESSyourself" target="_blank">here</a>.</p>
                    <p>* Find the publication <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007625" target="_blank">here</a> [1].</p>

                    <p>When I began graduate school at the University of Utah, and began considering working in the <a href="http://rutter.biochem.utah.edu/" target="_blank">Rutter lab</a>, Jared proposed a project to me which I found fascinating. He was interested in looking at protein translation using ribosome profilng [2], but in the context of metabolism of interest to the Rutter lab. I was hooked by the idea as it would allow a nice blend of wet lab and dry lab work. However, as I began exploring the data, I quickly found that the analysis side of things still had a lot of room for development, and there was no consensus dry methodology, and I realized that everyone designing their own scripts for processing these data could be problematic as much as it was inefficient. Additionally, key steps of data processing were missing from the open source community. Soon enough, I began thinking of ways I could help solve these issues and began work on <a href="https://github.com/XPRESSyourself" target="_blank">XPRESSyourself</a>.</p>

                    <br>
                    <h3>What does XPRESSyourself do?</h3>

                    <p><a href="https://github.com/XPRESSyourself" target="_blank">XPRESSyourself</a>'s goals are three-fold:</p>
                    <p>
                      1) Act as a reference implementation of ribosome profiling data analysis.
                      <br>
                      2) Provide automated tools missing from the ribosome profiling toolkit.
                      <br>
                      3) Where tools exist, help automate the tasks into a well-integrated pipeline.
                    </p>
                    <p><a href="https://github.com/XPRESSyourself" target="_blank">XPRESSyourself</a> is designed to require minimal inputs from the user and guide the user through the proper set-up of experimental variables to pass into the packages XPRESSyourself utilizes. XPRESSyourself is split into two packages: <a href="https://xpresspipe.readthedocs.io/en/latest/" target="_blank">XPRESSpipe</a> and <a href="https://xpressplot.readthedocs.io/en/latest/" target="_blank">XPRESSplot</a>. The former handles the processing and quality control of the raw FASTQ sequence files, and the latter handles normalization and analysis of the processed data. While the input requirements are detailed in the documentation and help menus, we tried to make this tool as approachable as possible. Therefore, we designed a sort of "choose your own adventure" style command builder (shown below) to walk the user through every consideration they would need to make for running <a href="https://xpresspipe.readthedocs.io/en/latest/" target="_blank">XPRESSpipe</a>, and <a href="https://github.com/XPRESSyourself/XPRESSplot/blob/main/example_notebook.ipynb" target="_blank">Jupyter Notebook</a> for example analyses for <a href="https://xpressplot.readthedocs.io/en/latest/" target="_blank">XPRESSplot</a>. By centralizing and automating the code needed to process this specialized data type, hopefully, reproducibility will increase and we will be able to extract the maximum amount of data from these labor-intensive and expensive experiments.
                    </p>

                    <center>
                      <p><img src="https://j-berg.github.io/blog/code/2020-01-19_XPRESSyourself/xpresspipe_build.png" style="position:relative;border-radius:0px;max-height:500px;"/></p>
                    </center>
                    <p align="center"><i>View of the XPRESSpipe command builder.</i></p>
                    <br>
                    <h3>Adding to the toolkit</h3>

                    <p>As mentioned earlier, when I began analyzing my own ribosome profiling data, I began to discover some key tools were missing from the toolkit. The most glaring vacancy in the toolkit centered around removing ribosome profiling-specific biases from the data. As the kinetics of translation initiation and termination are much slower than elongation dynamics [3], which is frequently the of most common interest to users of this technology, initation and termination can often provide a more murky picture of translation dynamics. This effect is further exacerbated by the use of cycloheximide during ribosome footprint harvesting. It is therefore recommended to disregard the first 45 nt and last 15 nt of each transcript, which are suspect to these kinetic consequences [4]. However, no open source tools exist for handling this problem. We decided to handle this issue during quantification of reads using a modified GTF reference file. However, as many transcripts in these records contain short exon 1 sections, often removing the first exon alone is not enough, and uneven truncation can have problematic consequences on quantification between transcripts (see image below). Therefore, we handle this truncation recursively.
                    </p>

                    <center>
                    <p><img src="https://j-berg.github.io/blog/code/2020-01-19_XPRESSyourself/truncation_cartoon.png" style="position:relative;border-radius:0px;max-height:400px;"/></p>
                    </center>
                    <p align="center"><i>The consequences of improper trimming.</i></p>

                    <p>Other features act to thread together steps of quality control that previously were not fully automated. Metagene read density, intron-specific gene-specific read coverage, read periodicity, and codon distributions are processed for each sample and a publication quality summary PDF is output so the user can quickly view and compare all samples to assess the quality of their ribosome profiling libraries.
                    </p>

                    <p>We designed XPRESSyourself to be as flexible as possible. Therefore, in addition to being able to processing ribosome profiling data (and its paired RNA-seq data for translation efficiency calculations), we have used it to analyze single-end RNA-seq, paired-end RNA-seq (we show an example in the supplement where we processed TCGA data), and more sequencing platforms. We have also used XPRESSplot to analyze even proteomics data.
                    </p>

                    <br>
                    <h3>Conclusions</h3>
                    <p>Certainly this is only meant to be a quick overview of the capabilities of <a href="https://github.com/XPRESSyourself" target="_blank">XPRESSyourself</a>. If you are interested in learning more about all the capabilities XPRESSyourself has to offer, I would suggest starting with the <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007625" target="_blank">paper</a> and then explore of the <a href="https://github.com/XPRESSyourself" target="_blank">documentation</a> if the tool looks useful to you. If you run into any challenges while using the software, feel free to reach out to me by submitting an <a href="https://github.com/XPRESSyourself/XPRESSpipe/issues" target="_blank">issue</a> on Github. I hope you find this tool useful! Happy analyzing!
                    </p>

                    <br>
                    <h3>References</h3>

                    1. Berg JA, Belyeu JR, Morgan JT, Ouyang Y, Bott AJ, Quinlan AR, Gertz J, Rutter J. <i>XPRESSyourself</i>: Enhancing, Standardizing, and Automating Ribosome Profiling Computational Analyses Yields Improved Insight into Data. (2020) <b><i>PLoS Comp. Biol.</i></b>
                    <br>
                    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                    <font size="2">
                      <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007625" target="_blank"><i>Publication</i></a>
                      &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                    </font>
                    <br>
                    <br>
                    2. Ingolia NT, Ghaemmaghami S, Newman JR, Weissman JS. Genome-wide analysis in vivo of translation with nucleotide resolution using ribosome profiling. (2009) Science, 324(5924), 218–223. <a href="https://science.sciencemag.org/content/324/5924/218" target="_blank">doi:10.1126/science.1168978</a>
                    <br>
                    <br>
                    3. Stein KC, Frydman J. The stop-and-go traffic regulating protein biogenesis: How translation kinetics controls proteostasis. (2019) The Journal of biological chemistry, 294(6), 2076–2084. <a href="http://www.jbc.org/content/294/6/2076" target="_blank">doi:10.1074/jbc.REV118.002814</a>
                    <br>
                    <br>
                    4. McGlincy NJ, Ingolia NT. Transcriptome-wide measurement of translation by ribosome profiling. (2017) Methods, 126, 112–129. <a href="https://www.sciencedirect.com/science/article/pii/S1046202316303292" target="_blank">doi:10.1016/j.ymeth.2017.05.028</a>
                    </div>

                    <br>
                    <br>


                    <div class="modal-footer d-flex justify-content-between">
                      <button type="button" class="btn" data-dismiss="modal">Close</button>
                      <div align="right">
                        <a href="https://github.com/j-berg/j-berg.github.io/tree/main/blog/code/2020-01-19_XPRESSyourself" target="_blank" class="m-btn">Code</a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <!-- project-1-modal end -->
 
          <!-- PROJECT 1 END -->
          <br>


          <!-- BLOG 0 START -->

          <div id="blog-0" class="col-12 m-60px-b md-m-30px-b">

            <a href="" class="p-btn" data-toggle="modal" data-target=".blog-0-modal">
              <h4>Understanding scaling in heatmaps</h4>
            </a>
            <p>(09 Apr 2019) A brief summary of the utility of plotting heatmaps using z-scored gene expression values in pattern identification.</p>

            <!-- blog-0-modal start  -->
            <div class="modal fade blog-0-modal" tabindex="-1" role="dialog" aria-labelledby="myLargeModalLabel" aria-hidden="true">
              <div class="modal-dialog modal-lg modal-dialog-scrollable">
                <div class="modal-content">
                  <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
                      <span aria-hidden="true">&times;</span>
                    </button>

                    <span class="post-meta"><b>09 Apr 2019</b>
                      <h4 class="modal-title" id="exampleModalScrollableTitle">Understanding scaling in heatmaps</h4>

                  </div>
                  <div class="modal-body">
                    <p>Heatmaps are an effective method for representing large swaths of data at once and uncovering underlying patterns within the data. Probably the most common use in biology is within high-throughput genomics experiments, such as RNA-seq. In RNA-seq experiments, one may want to uncover molecular patterns characteristic of a particular biological background, or track changes in gene expression across time. Whenever I make a heatmap, I always get the same question: <b><u>Why are you using Z-scores for the gene expression data? Why not use fold changes?</u></b> This blog post is intended as a blog post to summarize my answers to some common questions. I will be using examples relevant to RNA-seq, but the principles are applicable to other data types.</p>

                    <p>In the following examples, I will be borrowing heavily from a fantastic blog post by BIOMICKWATSON, titled <a href="http://www.opiniomics.org/you-probably-dont-understand-heatmaps/" target="_blank">"You probably don't understand heatmaps"</a>. The purpose of my blog post is not to steal the thunder of BIOMICKWATSON, but rather to offer a summary of questions I have encountered and an easy place to access this information (rather than me trying to explain myself over and over again).</p>

                    <br>
                    <h3>Why Z-score?</h3>

                    <p>Z-scoring (or standard scoring) of data essentially boils down to going gene by, calculating the mean of the the values across all samples for the gene, then determining how many standard deviations the actual value is from the mean. This is also commonly referred to as "mean centering." This is done using the following formula:</p>

                    $$ {z = (x – μ) / σ} $$

                    <p>Let's break this down a little:
                    <br>
                    - x: A given gene expression measurement (ex: 40)<br>
                    - μ: The mean of gene expression values across samples for a given gene (ex: 25)<br>
                    - σ: The standard deviation across samples for a given gene (ex: 5)<br>
                    - z: The outcoming z-score for each gene expression measurement</p>

                    <p>So in this example, the output z-score for this particular gene in this sample would be:</p>

                    $$ {3 = (40 – 25) / 5} $$

                    <p>So this gene having a z = 3 means its expression is 3 standard deviations above the mean. Pretty impressive, right?</p>

                    <center>
                      <p><img src="https://raw.githubusercontent.com/j-berg/j-berg.github.io/main/blog/code/2019-04-09_heatmaps/normal_dist_wikimedia.png" alt="See https://commons.wikimedia.org/wiki/File:The_Normal_Distribution.svg for license information. Source: https://upload.wikimedia.org/wikipedia/commons/thumb/2/25/The_Normal_Distribution.svg/1920px-The_Normal_Distribution.svg.png" style="position:relative;border-radius: 0px;"/></p>
                    </center>
                    <p align="center"><i>Scoring values along a normal distribution. Source: Wikipedia</i></p>

                    <br>
                    <h3>But Z-scores make no sense! Why not log<sub>2</sub>(fold change)?</h3>

                    <p>I get this response a lot, and I get it, Z-scoring gene expression may not be super intuitive. And I will admit, while Z-scores give you significance and direction, using solely Z-score misses an important piece of information: magnitude. But to plot a heatmap using using log<sub>2</sub>(fold change) also misses significance!</p>

                    <p>In cases where magnitude of change is important, I recommend people use a volcano plot to visualize their data. This captures magnitude, direction, and statistical significance, but will lose information like population variability (maybe have your experimental samples have one signature and the other half have another). You also cannot plot expression over a time course. But let's talk more about why Z-scores should be used in heatmaps, that's why we're here after all!</p>

                    <br>
                    <h3>Z-scores make each gene comparable to another</h3>

                    <p>Let's consider the following example from the aforementioned <a href="http://www.opiniomics.org/you-probably-dont-understand-heatmaps/" target="_blank">blog</a>.</p>

                    <p>In this first example, we have two gene expression patterns before any kind of gene normalization has been applied:</p>

                    <center>
                      
                      <p><img src="https://raw.githubusercontent.com/j-berg/j-berg.github.io/main/blog/code/2019-04-09_heatmaps/biomickwatson1.png" alt="Source: http://www.opiniomics.org/you-probably-dont-understand-heatmaps/" style="position:relative;border-radius: 0px;padding-top: 50px;"/></p>
                    </center>
                    <p align="center"><i>Source: BIOMICKWATSON</i></p>

                    <p><b>h1</b> and <b>l1</b> both follow the sample gene expression pattern. They appear to be under the same or similar regulation. Therefore, when performing unbiased clustering of these signatures, we would expect <b>h1</b> and <b>l1</b> to cluster together and <b>h2</b> and <b>l1</b> to cluster together. But let's look at the clustering outcome:</p>

                    <center>
                      <p><img src="https://raw.githubusercontent.com/j-berg/j-berg.github.io/main/blog/code/2019-04-09_heatmaps/biomickwatson2.png" alt="Source: http://www.opiniomics.org/you-probably-dont-understand-heatmaps/" style="position:relative;border-radius: 0px;"/></p>
                    </center>
                    <p align="center"><i>Source: BIOMICKWATSON</i></p>

                    <p>This result is a bit unexpected. Why did this happen? These patterns that should have emerged were hidden by the different magnitudes of expression, and so the clustering algorithm thought <b>h1</b> and <b>h2</b> shared a profile. This might lead researchers to make false conclusions concerning their data. Maybe based on this they determine <b>h1</b> and <b>h2</b> are in a pathway related to their disease and design drugs to target these two genes, when really the actually pathway involved <b>h1</b> and <b>h2</b> and they should have designed a different drug or one drug may have targeted both genes in a shared pathway and billions of dollars of R&D money was wasted instead.</p>

                    <p>Here is how the clustering should look when the data have been scaled:</p>
                    <center>
                      <p><img src="https://raw.githubusercontent.com/j-berg/j-berg.github.io/main/blog/code/2019-04-09_heatmaps/biomickwatson3.png" alt="Source: http://www.opiniomics.org/you-probably-dont-understand-heatmaps/" style="position:relative;border-radius: 0px;"/></p>
                    </center>
                    <p align="center"><i>Source: BIOMICKWATSON</i></p>

                    <br>
                    <h3>I don't care about clustering genes. Why should I still use Z-scores?</h3>

                    <p>Z-scores avoid swamping out patterns in the data. Let's take the following case for example:</p>

                    <center>
                      <p><img src="https://j-berg.github.io/blog/code/2019-04-09_heatmaps/heatmap_scaled_example.png" style="width:300px;position:relative; border-radius: 0px;"/></p>
                    </center>

                    <p>In the first case, everything looks pretty good. It seems like all gene expression levels decrease between wild-type and experimental cases, and this is easy to see when the data has been normalized in this manner. But let's look at the same exact dataset, but without normalization. One might say only GENE1 is really seeing any changes in gene expression, but nothing else is really going on.</p>

                    <center>
                      <p><img src="https://j-berg.github.io/blog/code/2019-04-09_heatmaps/heatmap_unscaled_example.png" style="width:300px;position:relative;border-radius: 0px;"/></p>
                    </center>

                    <p>When scaling has been applied, one can observe that all these genes appear to be under similar transcriptional control and interesting biology can be uncovered that would otherwise appear to be hidden.</p>

                    <p>To look at the code I used to create the last two figures, please refer <a href="https://github.com/j-berg/j-berg.github.io/tree/main/blog/code/2019-04-09_heatmaps" target="_blank">here</a>.</p>

                    <br>
                    <h3>References</h3>

                    1. BIOMICKWATSON. You probably don't understand heatmaps. <a href="http://www.opiniomics.org/you-probably-dont-understand-heatmaps" target="_blank">http://www.opiniomics.org/you-probably-dont-understand-heatmaps/</a>. Accessed: 2019-04-19.
                    <br><br>
                    2. Statistics How To. Z-Score: Definition, Formula and Calculation. <a href="https://www.statisticshowto.datasciencecentral.com/probability-and-statistics/z-score/#zscoreformulas" target="_blank">https://www.statisticshowto.datasciencecentral.com/probability-and-statistics/z-score/#zscoreformulas</a>. Accessed: 2019-04-19.
                    <br><br>
                    3. Wikipedia. Standard Score. <a href="https://en.wikipedia.org/wiki/Standard_score" target="_blank">https://en.wikipedia.org/wiki/Standard_score</a>. Accessed: 2019-04-19.


                    <br>
                    <br>

                  </div>
                  <div class="modal-footer d-flex justify-content-between">
                    <button type="button" class="btn" data-dismiss="modal">Close</button>

                  </div>
                </div>
              </div>
            </div>
            <!-- project-0-modal end -->
          </div>
          <!-- BLOG 0 END -->
          <br>




        </div> <!-- blog row -->
      </div> <!-- Container -->
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
      <br>
    </section>
    <!--  Blog End  -->
  </main>


  <!-- Footer-->
  <footer class="footer">
    <div class="container">
      <div class="row">
        <div class="col-12 text-center">
          <ul class="social-icons">
            <li><a href="https://twitter.com/jordanberg0" target="_blank"><i class="fab fa-twitter"></i></a></li>
            <li><a href="https://scholar.google.com/citations?user=0XsLZ3sAAAAJ&hl=en"><i class="fab fa-google"></i></a></li>
            <li><a href="https://github.com/j-berg"><i class="fab fa-github"></i></a></li>
          </ul>
        </div>
      </div>
      <p class="copyright">© 2020-2021 Jordan Berg. All Rights Reserved</p>
    </div>
  </footer>

  <!-- jQuery -->
  <script src="static/js/jquery-3.2.1.min.js"></script>
  <script src="static/js/jquery-migrate-3.0.0.min.js"></script>

  <!-- Plugins -->
  <script src="static/plugin/bootstrap/js/popper.min.js"></script>
  <script src="static/plugin/bootstrap/js/bootstrap.min.js"></script>
  <script src="static/plugin/owl-carousel/js/owl.carousel.min.js"></script>
  <script src="static/plugin/isotope/isotope.pkgd.min.js"></script>
  <script src="static/plugin/magnific/jquery.magnific-popup.min.js"></script>

  <!-- custom -->
  <script src="static/js/custom.js"></script>
  <script src="static/js/mail_send.js"></script>





</body>
<!-- Body End -->

</html>
