<!DOCTYPE html>
<html lang="en">

  <!-- jQuery -->
  <script src="../static/js/jquery-3.2.1.min.js"></script>
  <script src="../static/js/jquery-migrate-3.0.0.min.js"></script>

  <!-- Plugins -->
  <script src="../static/plugin/bootstrap/js/popper.min.js"></script>
  <script src="../static/plugin/bootstrap/js/bootstrap.min.js"></script>
  <script src="../static/plugin/owl-carousel/js/owl.carousel.min.js"></script>
  <script src="../static/plugin/isotope/isotope.pkgd.min.js"></script>
  <script src="../static/plugin/magnific/jquery.magnific-popup.min.js"></script>

  <!-- custom -->
  <script src="../static/js/custom.js"></script>
  <script src="../static/js/mail_send.js"></script>
<head>

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-1QG2TZS458"></script>
  <script>
    window.dataLayer = window.dataLayer || [];

    function gtag() {
      dataLayer.push(arguments);
    }
    gtag('js', new Date());

    gtag('config', 'G-1QG2TZS458');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta name="description" content="Website for Jordan A. Berg, Biochemistry Ph.D. student at the University of Utah">
  <meta name="keywords" content="Jordan, Berg, Biochemistry, Utah, Computational, Biology, Metabolism">
  <meta name="author" content="Jordan Berg">
  <meta name="google-site-verification" content="7R5dCQpyVk7BtHBQhqaXIUbCffeulxE92qqUJDMj7nM" />

  <!---Font Icon-->
  <link href="../static/plugin/font-awesome/css/fontawesome-all.min.css" rel="stylesheet">
  <link href="../static/plugin/themify-icons/themify-icons.css" rel="stylesheet">
  <!-- / -->

  <!-- Plugin CSS -->
  <link href="../static/plugin/bootstrap/css/bootstrap.min.css" rel="stylesheet">
  <link href="../static/plugin/owl-carousel/css/owl.carousel.min.css" rel="stylesheet">
  <link href="../static/plugin/magnific/magnific-popup.css" rel="stylesheet">
  <!-- / -->

  <!-- Theme Style -->
  <link href="../static/css/styles.css" rel="stylesheet">
  <link href="../static/css/color/color-7.css" rel="stylesheet" id="color_theme">
  <!-- / -->

  <!-- Favicon -->
  <link rel="icon" href="../static/img/Noun_project_network_icon_1365244_cc.svg" />
  <!-- / -->

  <!-- Custom Style -->
  <link href="../static/css/custom.css" rel="stylesheet">
  <!-- / -->

</head>

<!-- Body Start -->

<body data-spy="scroll" data-target="#navbarRyan" data-offset="98">

  <!-- Loading -->
  <div id="loading">
    <div class="load-circle"><span class="one"></span></div>
  </div>
  <!-- / -->

  <!-- Header -->
  <header>
    <nav class="navbar header-nav header-nav-white fixed-top navbar-expand-lg">
      <div class="container-fluid">
        <!-- Brand -->
        <a class="navbar-brand font-alt" href="index.html">Network Metabolism Group</a>
        <!-- / -->

        <!-- Mobile Toggle -->
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarRyan" aria-controls="navbarRyan" aria-expanded="false" aria-label="Toggle navigation">
          <span></span>
          <span></span>
          <span></span>
        </button>
        <!-- / -->

        <!-- Top Menu -->
        <div class="collapse navbar-collapse justify-content-end" id="navbarRyan">
          <ul class="navbar-nav">
            <li><a class="nav-link" href="../index.html">About</a></li>
            <li><a class="nav-link" href="../research.html">Research</a></li>
            <li><a class="nav-link" href="../publications.html">Publications</a></li>
            <li><a class="nav-link" href="../people.html">People</a></li>
            <li><a class="nav-link" href="../funding.html">Funding</a></li>
            <li><a class="nav-link" href="../blog.html">Blog</a></li>
            <li><a class="nav-link" href="../contact.html">Contact</a></li>
            <li></li>
            <li></li>
          </ul>
        </div>
        <!-- / -->

      </div><!-- Container -->
    </nav> <!-- Navbar -->
  </header>
  <!-- Header End -->

  <!-- Page Title -->
  <script>
    /* --------------------START EDITING HERE-------------------- */
    var blog_items = {
        page: "blog/2019-04-09_understanding-scaling-in-heatmaps.html",
        title: "Understanding scaling in heatmaps",
        date: "09 Apr 2019",
        tags: ["heatmaps", "scaling"],
        summary: "A brief summary of the utility of plotting heatmaps using z-scored gene expression values in pattern identification.",
        entry: ""
    };
    /* --------------------STOP EDITING HERE--------------------- */

    window.onload = function () {
      document.title = blog_items.title;
      document.getElementById("entry-title").innerHTML = blog_items.title;
      let this_header = "<h6>" + blog_items.date + " by Jordan Berg</h6>";
      for (let t in blog_items.tags) {
        this_header = this_header + 
          "<a href='../tags.html?" + blog_items.tags[t] + "'><div class='topic-tag'>" + blog_items.tags[t] + "</div></a>";
      }
      this_header = this_header +
        "<blockquote class='custom-blockquote'>" + blog_items.summary + "</blockquote>" +
        "<br><br>";
      document.getElementById("entry-header").innerHTML = this_header
    };
  </script>

  <!-- / -->

  <!-- Page Content -->
  <main class="gray-bg">
    <section id="blog" class="section">
      <div class="container">
        <div class="row justify-content-center m-60px-b md-m-30px-b">
            <div class="section-title text-center">
                <h2 id="entry-title" class="font-alt"></h2>
            </div>
        </div>

        <div class="row md-m-30px-b md-m-10px-r md-m-10px-l">
          <div id="entry-header"></div>
          <div id="entry-content">
            <!-- --------------------START EDITING HERE-------------------- -->
            <p>Heatmaps are an effective method for representing large swaths of data at once and uncovering underlying patterns within the data. Probably the most common use in biology is within high-throughput genomics experiments, such as RNA-seq. In RNA-seq experiments, one may want to uncover molecular patterns characteristic of a particular biological background, or track changes in gene expression across time. Whenever I make a heatmap, I always get the same question: <b><u>Why are you using Z-scores for the gene expression data? Why not use fold changes?</u></b> This blog post is intended as a blog post to summarize my answers to some common questions. I will be using examples relevant to RNA-seq, but the principles are applicable to other data types.</p>

            <p>In the following examples, I will be borrowing heavily from a fantastic blog post by BIOMICKWATSON, titled <a href="http://www.opiniomics.org/you-probably-dont-understand-heatmaps/" target="_blank">"You probably don't understand heatmaps"</a>. The purpose of my blog post is not to steal the thunder of BIOMICKWATSON, but rather to offer a summary of questions I have encountered and an easy place to access this information (rather than me trying to explain myself over and over again).</p>

            <br>
            <h3>Why Z-score?</h3>

            <p>Z-scoring (or standard scoring) of data essentially boils down to going gene by, calculating the mean of the the values across all samples for the gene, then determining how many standard deviations the actual value is from the mean. This is also commonly referred to as "mean centering." This is done using the following formula:</p>

            $$ {z = (x – μ) / σ} $$

            <p>Let's break this down a little:
            <br>
            - x: A given gene expression measurement (ex: 40)<br>
            - μ: The mean of gene expression values across samples for a given gene (ex: 25)<br>
            - σ: The standard deviation across samples for a given gene (ex: 5)<br>
            - z: The outcoming z-score for each gene expression measurement</p>

            <p>So in this example, the output z-score for this particular gene in this sample would be:</p>

            $$ {3 = (40 – 25) / 5} $$

            <p>So this gene having a z = 3 means its expression is 3 standard deviations above the mean. Pretty impressive, right?</p>

            <center>
                <p><img src="https://raw.githubusercontent.com/j-berg/j-berg.github.io/main/blog/code/2019-04-09_heatmaps/normal_dist_wikimedia.png" alt="See https://commons.wikimedia.org/wiki/File:The_Normal_Distribution.svg for license information. Source: https://upload.wikimedia.org/wikipedia/commons/thumb/2/25/The_Normal_Distribution.svg/1920px-The_Normal_Distribution.svg.png" style="position:relative;border-radius: 0px;"/></p>
            </center>
            <p align="center"><i>Scoring values along a normal distribution. Source: Wikipedia</i></p>

            <br>
            <h3>But Z-scores make no sense! Why not log<sub>2</sub>(fold change)?</h3>

            <p>I get this response a lot, and I get it, Z-scoring gene expression may not be super intuitive. And I will admit, while Z-scores give you significance and direction, using solely Z-score misses an important piece of information: magnitude. But to plot a heatmap using using log<sub>2</sub>(fold change) also misses significance!</p>

            <p>In cases where magnitude of change is important, I recommend people use a volcano plot to visualize their data. This captures magnitude, direction, and statistical significance, but will lose information like population variability (maybe have your experimental samples have one signature and the other half have another). You also cannot plot expression over a time course. But let's talk more about why Z-scores should be used in heatmaps, that's why we're here after all!</p>

            <br>
            <h3>Z-scores make each gene comparable to another</h3>

            <p>Let's consider the following example from the aforementioned <a href="http://www.opiniomics.org/you-probably-dont-understand-heatmaps/" target="_blank">blog</a>.</p>

            <p>In this first example, we have two gene expression patterns before any kind of gene normalization has been applied:</p>

            <center>
                
                <p><img src="https://raw.githubusercontent.com/j-berg/j-berg.github.io/main/blog/code/2019-04-09_heatmaps/biomickwatson1.png" alt="Source: http://www.opiniomics.org/you-probably-dont-understand-heatmaps/" style="position:relative;border-radius: 0px;padding-top: 50px;"/></p>
            </center>
            <p align="center"><i>Source: BIOMICKWATSON</i></p>

            <p><b>h1</b> and <b>l1</b> both follow the sample gene expression pattern. They appear to be under the same or similar regulation. Therefore, when performing unbiased clustering of these signatures, we would expect <b>h1</b> and <b>l1</b> to cluster together and <b>h2</b> and <b>l1</b> to cluster together. But let's look at the clustering outcome:</p>

            <center>
                <p><img src="https://raw.githubusercontent.com/j-berg/j-berg.github.io/main/blog/code/2019-04-09_heatmaps/biomickwatson2.png" alt="Source: http://www.opiniomics.org/you-probably-dont-understand-heatmaps/" style="position:relative;border-radius: 0px;"/></p>
            </center>
            <p align="center"><i>Source: BIOMICKWATSON</i></p>

            <p>This result is a bit unexpected. Why did this happen? These patterns that should have emerged were hidden by the different magnitudes of expression, and so the clustering algorithm thought <b>h1</b> and <b>h2</b> shared a profile. This might lead researchers to make false conclusions concerning their data. Maybe based on this they determine <b>h1</b> and <b>h2</b> are in a pathway related to their disease and design drugs to target these two genes, when really the actually pathway involved <b>h1</b> and <b>h2</b> and they should have designed a different drug or one drug may have targeted both genes in a shared pathway and billions of dollars of R&D money was wasted instead.</p>

            <p>Here is how the clustering should look when the data have been scaled:</p>
            <center>
                <p><img src="https://raw.githubusercontent.com/j-berg/j-berg.github.io/main/blog/code/2019-04-09_heatmaps/biomickwatson3.png" alt="Source: http://www.opiniomics.org/you-probably-dont-understand-heatmaps/" style="position:relative;border-radius: 0px;"/></p>
            </center>
            <p align="center"><i>Source: BIOMICKWATSON</i></p>

            <br>
            <h3>I don't care about clustering genes. Why should I still use Z-scores?</h3>

            <p>Z-scores avoid swamping out patterns in the data. Let's take the following case for example:</p>

            <center>
                <p><img src="https://j-berg.github.io/blog/code/2019-04-09_heatmaps/heatmap_scaled_example.png" style="width:300px;position:relative; border-radius: 0px;"/></p>
            </center>

            <p>In the first case, everything looks pretty good. It seems like all gene expression levels decrease between wild-type and experimental cases, and this is easy to see when the data has been normalized in this manner. But let's look at the same exact dataset, but without normalization. One might say only GENE1 is really seeing any changes in gene expression, but nothing else is really going on.</p>

            <center>
                <p><img src="https://j-berg.github.io/blog/code/2019-04-09_heatmaps/heatmap_unscaled_example.png" style="width:300px;position:relative;border-radius: 0px;"/></p>
            </center>

            <p>When scaling has been applied, one can observe that all these genes appear to be under similar transcriptional control and interesting biology can be uncovered that would otherwise appear to be hidden.</p>

            <p>To look at the code I used to create the last two figures, please refer <a href="https://github.com/j-berg/j-berg.github.io/tree/main/blog/code/2019-04-09_heatmaps" target="_blank">here</a>.</p>

            <br>
            <h3>References</h3>

            1. BIOMICKWATSON. You probably don't understand heatmaps. <a href="http://www.opiniomics.org/you-probably-dont-understand-heatmaps" target="_blank">http://www.opiniomics.org/you-probably-dont-understand-heatmaps/</a>. Accessed: 2019-04-19.
            <br><br>
            2. Statistics How To. Z-Score: Definition, Formula and Calculation. <a href="https://www.statisticshowto.datasciencecentral.com/probability-and-statistics/z-score/#zscoreformulas" target="_blank">https://www.statisticshowto.datasciencecentral.com/probability-and-statistics/z-score/#zscoreformulas</a>. Accessed: 2019-04-19.
            <br><br>
            3. Wikipedia. Standard Score. <a href="https://en.wikipedia.org/wiki/Standard_score" target="_blank">https://en.wikipedia.org/wiki/Standard_score</a>. Accessed: 2019-04-19.


            <br>
            <br>



            <!-- --------------------STOP EDITING HERE--------------------- -->
            <br>
            <br>
            <br>
            <br>
            <div id="disqus_thread"></div>
            <script>
                /**
                *  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
                *  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables    */
                
                var disqus_config = function () {  
                  this.page.url = "https://j-berg.github.io/" + blog_items.page;  // Replace PAGE_URL with your page's canonical URL variable
                  this.page.title = blog_items.title;
                };
          
                (function() { // DON'T EDIT BELOW THIS LINE
                var d = document, s = d.createElement('script');
                s.src = 'https://j-berg.disqus.com/embed.js';
                s.setAttribute('data-timestamp', +new Date());
                (d.head || d.body).appendChild(s);
                })();
            </script>
            <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
          </div>
        </div>
      </div>
    </div> <!-- blog row -->
  </div> <!-- Container -->
  <br>
  <br>
  <br>
  <br>
</section>
<!--  Blog End  -->
</main>


<!-- Footer-->
<footer class="footer">
<div class="container">
  <div class="row">
    <div class="col-12 text-center">
      <ul class="social-icons">
        <li><a href="https://twitter.com/jordanberg0" target="_blank"><i class="fab fa-twitter"></i></a></li>
        <li><a href="https://scholar.google.com/citations?user=0XsLZ3sAAAAJ&hl=en"><i class="fab fa-google"></i></a></li>
        <li><a href="https://github.com/j-berg"><i class="fab fa-github"></i></a></li>
      </ul>
    </div>
  </div>
  <p class="copyright">© 2020-2021 Jordan Berg. All Rights Reserved</p>
</div>
</footer>


</body>
</html>